### PR TITLE
add agent 6 schedule to create rc pr workflow

### PR DESCRIPTION
**Pull request by sabrina-datadog**
### What does this PR do?
Add a weekly schedule to the create rc pr workflow for agent 6 release candidates. [ACIX-454](https://datadoghq.atlassian.net/browse/ACIX-454)
- change `find_release_branches` job to return `[6.53.x]` when it is for a6
- use `--patch-version` for `inv release.create-rc` to bump to the right version after a final version (`6.53.x-rc-1`)
- skip `yes_no_question` user interaction because it is a github workflow
- reuse the `set_git_config` function from this [OTel PR](https://github.com/DataDog/datadog-agent/pull/31545/files#diff-909ceda28445b406c806e55409c5e4853ba7f58c2bb4e1b4ef43c3f781a7606dR509)

### Motivation
We want to ensure that agent 6 can be released at any point. So a weekly rc will be run to test that the agent 6 pipeline works. 

### Describe how you validated your changes

created a [test branch](https://github.com/DataDog/datadog-agent/compare/sabrina/a6-workflow...sabrina/a6-workflow-test) to simulate when:
 - it is a weekly agent 6 rc cron job: [PR](https://github.com/DataDog/datadog-agent/pull/32078) that was created from [successful workflow](https://github.com/DataDog/datadog-agent/actions/runs/12290204973)
 - it is a agent 7 rc cron job: [PR](https://github.com/DataDog/datadog-agent/pull/32077) that was created from [successful workflow](https://github.com/DataDog/datadog-agent/actions/runs/12290082901)

### Additional Notes

[ACIX-454]: https://datadoghq.atlassian.net/browse/ACIX-454?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .github/workflows/create_rc_pr.yml [hunk 1]
```diff
@@ -87,17 +96,18 @@ jobs:
...
-                  echo "CHANGES=$(inv -e release.check-for-changes -r "$MATRIX" "$WARNING")" >> $GITHUB_OUTPUT
+                  echo "CHANGES=$(inv -e release.check-for-changes -r "$MATRIX" "$WARNING" | tail -n 1)" >> $GITHUB_OUTPUT
                 else
-                  echo "CHANGES=$(inv -e release.check-for-changes -r "$MATRIX")" >> $GITHUB_OUTPUT
+                  echo "CHANGES=$(inv -e release.check-for-changes -r "$MATRIX" | tail -n 1)" >> $GITHUB_OUTPUT
                 fi
 
             - name: Create RC PR
-              if: ${{ steps.check_for_changes.outputs.CHANGES == 'true'}}
+              if: ${{ steps.check_for_changes.outputs.CHANGES == 'true' || env.IS_AGENT6_RELEASE == 'true' }}
```
Comment (kacper-murzyn) [inline .github/workflows/create_rc_pr.yml:105]: Wouldn't this condition mean that every time we run this action on the schedule dedicated for Agent 6, then we will try to create new RC, regardless if there are any new changes? Because `env.IS_AGENT6_RELEASE` is always true in this case.

Comment (CelianR) [inline .github/workflows/create_rc_pr.yml:105]: Should we remove the previous step if the output is ignored when releasing agent 6 ?

File: .github/workflows/create_rc_pr.yml [hunk 2]
```diff
@@ -87,17 +96,18 @@ jobs:
...
               env:
                 MATRIX: ${{ matrix.value }}
               run: |
-                git config user.name "github-actions[bot]"
-                git config user.email "41898282+github-actions[bot]@users.noreply.github.com"
-                git fetch
-                inv -e release.create-rc "$MATRIX" --slack-webhook=${{ secrets.AGENT_RELEASE_SYNC_SLACK_WEBHOOK }}
+                if ${{ env.IS_AGENT6_RELEASE == 'true' }}; then
+                  inv -e release.create-rc "$MATRIX" --slack-webhook=${{ secrets.AGENT_RELEASE_SYNC_SLACK_WEBHOOK }} --github-action --patch-version
```
Comment (CelianR) [inline .github/workflows/create_rc_pr.yml:110]:
```suggestion
                  inv -e release.create-rc -r "$MATRIX" --slack-webhook=${{ secrets.AGENT_RELEASE_SYNC_SLACK_WEBHOOK }} --github-action --patch-version
```
Following https://github.com/DataDog/datadog-agent/pull/32054

File: .github/workflows/create_rc_pr.yml [hunk 3]
```diff
@@ -87,17 +96,18 @@ jobs:
...
+                else
+                  inv -e release.create-rc "$MATRIX" --slack-webhook=${{ secrets.AGENT_RELEASE_SYNC_SLACK_WEBHOOK }} --github-action
```
Comment (CelianR) [inline .github/workflows/create_rc_pr.yml:112]:
```suggestion
                  inv -e release.create-rc -r "$MATRIX" --slack-webhook=${{ secrets.AGENT_RELEASE_SYNC_SLACK_WEBHOOK }} --github-action
```
Following https://github.com/DataDog/datadog-agent/pull/32054

File: tasks/libs/common/git.py [hunk 4]
```diff
@@ -270,3 +275,20 @@ def get_last_release_tag(ctx, repo, pattern):
...
+def set_git_config(key, value):
+    subprocess.run(['git', 'config', key, value])
+
+
+def revert_git_config(original_config):
+    for key, value in original_config.items():
+        if value is None:
+            subprocess.run(['git', 'config', '--unset', key])
+        else:
+            subprocess.run(['git', 'config', key, value])
```
Comment (CelianR) [inline tasks/libs/common/git.py:294]: **💬 suggestion:** ‏Could you use `ctx.run` instead? (e.g. `ctx.run('...', hide=True).stdout`). Note that this will raise an error if the process exits with non zero value

File: tasks/release.py [hunk 5]
```diff
@@ -427,12 +428,16 @@ def create_rc(ctx, release_branch, patch_version=False, upstream="origin", slack
         This also requires that there are no local uncommitted changes, that the current branch is 'main' or the
         release branch, and that no branch named 'release/<new rc version>' already exists locally or upstream.
     """
-
     major_version = get_version_major(release_branch)
 
     with agent_context(ctx, release_branch):
         github = GithubAPI(repository=GITHUB_REPO_NAME)
 
+        if github_action:
```
Comment (CelianR) [inline tasks/release.py:436]: **🥜 nitpick:** ‏Maybe this can be deduced via environment variables?

**Review by CelianR — APPROVED** [review]

File: tasks/release.py [hunk 6]
```diff
@@ -427,12 +428,16 @@ def create_rc(ctx, release_branch, patch_version=False, upstream="origin", slack
...
+            set_git_config('user.name', 'github-actions[bot]')
+            set_git_config('user.email', 'github-actions[bot]@users.noreply.github.com')
+            upstream = f"https://x-access-token:{os.environ.get('GITHUB_TOKEN')}@github.com/{GITHUB_REPO_NAME}.git"
```
Comment (chouquette) [inline tasks/release.py:439]: Are we sure this will not leak the token when executing the command?

File: tasks/libs/common/git.py [hunk 7]
```diff
@@ -179,7 +180,11 @@ def try_git_command(ctx, git_command):
                     "orange",
                 )
             )
-            do_retry = yes_no_question("Do you want to retry this operation?", color="orange", default=True)
+            do_retry = (
+                False
+                if github_action
+                else yes_no_question("Do you want to retry this operation?", color="orange", default=True)
+            )
```
Comment (chouquette) [inline tasks/libs/common/git.py:187]: It feels like this could be replaced by checking if a TTY is attached rather that "if running under a CI"

File: tasks/libs/common/git.py [hunk 8]
```diff
@@ -270,3 +275,20 @@ def get_last_release_tag(ctx, repo, pattern):
...
     return last_tag_commit, last_tag_name
+
+
+def get_git_config(key):
+    result = subprocess.run(['git', 'config', '--get', key], capture_output=True, text=True)
+    return result.stdout.strip() if result.returncode == 0 else None
+
+
+def set_git_config(key, value):
+    subprocess.run(['git', 'config', key, value])
```
Comment (chouquette) [inline tasks/libs/common/git.py:286]: These helpers are a good idea! They could even be split in their own PR and be used throughout our tasks if this PR takes long to be merged (but definitely not required)

**Review by songy23 — APPROVED** [review]
Approval for OTel

File: tasks/release.py (hunk 6 above)
Comment (sabrina-datadog) [reply tasks/release.py:439]: GitHub masks the secret by default, so the token isn't exposed ([example](https://github.com/DataDog/datadog-agent/actions/runs/12290204973/job/34296969010#step:6:750)). let me know if there are any concerns

File: tasks/libs/common/git.py (hunk 8 above)
Comment (sabrina-datadog) [reply tasks/libs/common/git.py:286]: I will make a ticket to update the rest of our tasks

File: .github/workflows/create_rc_pr.yml (hunk 1 above)
Comment (kacper-murzyn) [reply .github/workflows/create_rc_pr.yml:105]: I see. Then it all makes sense 👍

File: tasks/libs/common/git.py (hunk 4 above)
Comment (sabrina-datadog) [reply tasks/libs/common/git.py:294]: I'm trying to reuse what the OTel folks did [here](https://github.com/DataDog/datadog-agent/pull/31545/files#diff-909ceda28445b406c806e55409c5e4853ba7f58c2bb4e1b4ef43c3f781a7606dR509), following @chouetz's offline suggestion

File: .github/workflows/create_rc_pr.yml (hunk 1 above)
Comment (sabrina-datadog) [reply .github/workflows/create_rc_pr.yml:99]: No, I think we should keep it because the `check_for_changes` function not only checks for changes but can also modifies the tags. There is additional logic that needs to be executed as part of the process [here](https://github.com/DataDog/datadog-agent/blob/main/tasks/release.py#L1214-L1219).

File: .github/workflows/create_rc_pr.yml [hunk 9]
```diff
@@ -19,18 +21,21 @@ jobs:
...
 
             - name: Install python
+              if: ${{ env.IS_AGENT6_RELEASE == 'false' }}
               uses: actions/setup-python@0b93645e9fea7318ecaed2b359559ac225c90a2b # v5.3.0
               with:
                 python-version: 3.11
                 cache: "pip"
 
             - name: Install Python dependencies
+              if: ${{ env.IS_AGENT6_RELEASE == 'false' }}
```
[the remark this answers is not in the document]
Comment (sabrina-datadog) [reply .github/workflows/create_rc_pr.yml:38]: Since the agent 6 will only have one unrelease branch (`6.53.x`) there is no need to find them. I tried adding an if condition at the job level, but it doesn't have access to the `env` variables at that point ([docs](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/accessing-contextual-information-about-workflow-runs#context-availability)).

**Review by spencergilbert — APPROVED** [review]

File: tasks/release.py [hunk 10]
```diff
@@ -432,11 +433,16 @@ def create_rc(ctx, release_branch, patch_version=False, upstream="origin", slack
         This also requires that there are no local uncommitted changes, that the current branch is 'main' or the
         release branch, and that no branch named 'release/<new rc version>' already exists locally or upstream.
     """
-
     major_version = get_version_major(release_branch)
 
     with agent_context(ctx, release_branch):
         github = GithubAPI(repository=GITHUB_REPO_NAME)
+        github_action = os.environ.get("GITHUB_ACTIONS")
```
[the remark this answers is not in the document]
Comment (sabrina-datadog) [reply tasks/release.py:440]: yea, this is the [documentation](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/store-information-in-variables#default-environment-variables) for the environment variable:

image
